src/MagnumPlugins/AssimpImporter/AssimpImporter.cpp

```diff
@@ -0,0 +1,526 @@
+*/
+
+#include "AssimpImporter.h"
+
+#include <algorithm>
+#include <cstring> /* std::memcpy */
+#include <limits>
+#include <unordered_map>
+#include <Corrade/Containers/ArrayView.h>
+#include <Corrade/Utility/Directory.h>
+#include <Magnum/Mesh.h>
+#include <Magnum/Math/Quaternion.h>
+#include <Magnum/Math/Vector.h>
+#include <Magnum/PixelFormat.h>
+#include <Magnum/PixelStorage.h>
+#include <Magnum/Trade/CameraData.h>
+#include <Magnum/Trade/ImageData.h>
+#include <Magnum/Trade/LightData.h>
+#include <Magnum/Trade/MeshData3D.h>
+#include <Magnum/Trade/MeshObjectData3D.h>
+#include <Magnum/Trade/PhongMaterialData.h>
+#include <Magnum/Trade/SceneData.h>
+#include <Magnum/Trade/TextureData.h>
+
+#include <MagnumPlugins/AnyImageImporter/AnyImageImporter.h>
+
+#include <assimp/postprocess.h>
+
+namespace Magnum { namespace Math { namespace Implementation {
+
+template<> struct VectorConverter<3, Float, aiColor3D> {
+    static Vector<3, Float> from(const aiColor3D& other) {
+        return {other.r, other.g, other.b};
+    }
+
+    static aiColor3D to(const Vector<3, Float>& other) {
+        return {other[0], other[1], other[2]};
+    }
+};
+
+}}}
+
+namespace Magnum { namespace Trade {
+
+using namespace Magnum::Math::Literals;
+
+AssimpImporter::AssimpImporter() = default;
+
+AssimpImporter::AssimpImporter(PluginManager::Manager<AbstractImporter>& manager): AbstractImporter(manager) {}
+
+AssimpImporter::AssimpImporter(PluginManager::AbstractManager& manager, const std::string& plugin): AbstractImporter(manager, plugin) {}
+
+AssimpImporter::~AssimpImporter() = default;
+
+auto AssimpImporter::doFeatures() const -> Features { return Feature::OpenData; }
+
+bool AssimpImporter::doIsOpened() const { return !!_scene; }
+
+void AssimpImporter::doOpenData(const Containers::ArrayView<const char> data) {
+    _scene = _importer.ReadFileFromMemory(data.data(), data.size(), aiProcess_Triangulate | aiProcess_SortByPType);
+
+    if(_scene) {
+        createNodeIndices();
+        createMaterialIndices();
+    }
+}
+
+void AssimpImporter::doOpenFile(const std::string& filename) {
+    _scene = _importer.ReadFile(filename, aiProcess_Triangulate | aiProcess_SortByPType);
+
+    /* If succeeded, save file path for later */
+    if (_scene) {
+        _filePath = Utility::Directory::path(filename);
+        createNodeIndices();
+        createMaterialIndices();
+    }
+}
+
+void AssimpImporter::doClose() {
+    _importer.FreeScene();
+    _scene = nullptr;
+    _filePath = std::nullopt;
+
+    _nodes.clear();
+    _nodeIndices.clear();
+    _nodeInstances.clear();
+    _materialIndicesForName.clear();
+    _textureIndices.clear();
+}
+
+Int AssimpImporter::doDefaultScene() { return 0; }
+
+UnsignedInt AssimpImporter::doSceneCount() const { return _scene->mRootNode ? 1 : 0; }
+
+std::optional<SceneData> AssimpImporter::doScene(UnsignedInt) {
+    const aiNode* root = _scene->mRootNode;
+    return SceneData{{}, {_nodeIndices[root]}, root};
```

Comment:
The root node should be equivalent to the scene node, so SceneData should contain directly the children of the root node.